hpat/tests/test_series.py

```diff
@@ -1809,13 +1809,103 @@ def test_impl(S, param_skipna):
             result = hpat_func(S, param_skipna)
             self.assertEqual(result, result_ref)
 
-    def test_series_value_counts(self):
+    def test_series_value_counts_int(self):
         def test_impl(S):
             return S.value_counts()
+
+        additional_data = [1, 2, 3, 1, 1, 3]
+
         hpat_func = hpat.jit(test_impl)
 
-        S = pd.Series(['AA', 'BB', 'C', 'AA', 'C', 'AA'])
-        pd.testing.assert_series_equal(hpat_func(S), test_impl(S))
+        for data in test_global_input_data_integer64:
+            S = pd.Series(data + additional_data)
+            # Remove sort_index() after implementing sorting with the same number of frequency
+            pd.testing.assert_series_equal(hpat_func(S).sort_index(), test_impl(S).sort_index())
+
+    def test_series_value_counts_float(self):
+        def test_impl(S):
+            return S.value_counts()
+
+        additional_data = [0.1, 0., 0.1, 0.1]
+
+        hpat_func = hpat.jit(test_impl)
+
+        for data in test_global_input_data_float64:
+            S = pd.Series(data + additional_data)
+            # Remove sort_index() after implementing sorting with the same number of frequency
+            pd.testing.assert_series_equal(hpat_func(S).sort_index(), test_impl(S).sort_index())
+
+    def test_series_value_counts_sort(self):
+        def test_impl(S, asceding):
+            return S.value_counts(sort=True, ascending=asceding)
+
+        hpat_func = hpat.jit(test_impl)
+
+        data = [1, 0, 0, 1, 1, -1, 0, -1, 0]
+
+        for asceding in (False, True):
+            S = pd.Series(data)
+            pd.testing.assert_series_equal(hpat_func(S, asceding), test_impl(S, asceding))
+
+    @unittest.skip('Unimplemented: need handling of numpy.nan comparison')
+    def test_series_value_counts_dropna_false(self):
+        def test_impl(S):
+            return S.value_counts(dropna=False)
+
+        data_to_test = [[1, 2, 3, 1, 1, 3],
+                        [1, 2, 3, np.nan, 1, 3, np.nan, np.inf],
+                        [0.1, 3., np.nan, 3., 0.1, 3., np.nan, np.inf, 0.1, 0.1]]
+
+        hpat_func = hpat.jit(test_impl)
+
+        for data in data_to_test:
+            S = pd.Series(data)
+            pd.testing.assert_series_equal(hpat_func(S), test_impl(S))
+
+    def test_series_value_counts_str(self):
```

Comment:
Tests `test_series_value_counts_str` and `test_series_value_counts_str_sort_ascending` also can be combined if to parameterize `test_impl(S, sort, ascending)`.